gio/src/task.rs

```diff
@@ -60,6 +60,51 @@ impl Task {
         }
     }
 
+    #[doc(alias = "g_task_run_in_thread")]
+    pub fn run_in_thread<Q>(&self, task_func: Q)
+    where
+        Q: FnOnce(&Self, Option<&glib::Object>, Option<&Cancellable>),
+        Q: Send + 'static,
+    {
+        let task_func_data = Box_::new(task_func);
+
+        // FIXME: we store the func pointer into the task data
+        // for now this is okaish, since we do not expose a way to
+        // set the task data in the bindings
+        unsafe {
+            ffi::g_task_set_task_data(
```

Comment:
If `run_in_thread()` is called multiple times, this will probably cause interesting problems? Or is that actually OK?

Reply:
I do not think you can call `run_in_thread` multiple times, no?

Reply:
What's preventing it from being called twice currently? :)

Reply:
mmm... it never even occurred to me to call it multiple times even from the C api, but you are right.

Any suggestion on how to enforce this?

Reply:
Maybe check if there is task data, and if there is panic